scm-ui/ui-webapp/src/containers/InitializationAdminAccountStep.tsx

```diff
@@ -0,0 +1,227 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import React, { FC, FormEvent, useEffect, useState } from "react";
+import { apiClient, validation, ErrorNotification, InputField, SubmitButton } from "@scm-manager/ui-components";
+import { useTranslation } from "react-i18next";
+import styled from "styled-components";
+import { useMutation } from "react-query";
+import { isDisplayNameValid, isPasswordValid } from "../users/components/userValidation";
+import { Links, Link } from "@scm-manager/ui-types";
+
+const HeroSection = styled.section`
+  padding-top: 2em;
+`;
+
+type Props = {
+  data: { _links: Links };
+};
+
+type AdminAccountCreation = {
+  startupToken: string;
+  userName: string;
+  displayName: string;
+  email: string;
+  password: string;
+  passwordConfirmation: string;
+};
+
+const createAdmin = (link: string) => {
+  return (data: AdminAccountCreation) => {
+    return apiClient.post(link, data, "application/json").then(() => {
+      return new Promise<void>((resolve) => resolve());
+    });
+  };
+};
+
+const useCreateAdmin = (link: string) => {
+  const { mutate, isLoading, error, isSuccess } = useMutation<void, Error, AdminAccountCreation>(createAdmin(link));
+  return {
+    create: mutate,
+    isLoading,
+    error,
+    isCreated: isSuccess,
+  };
+};
+
+const InitializationAdminAccountStep: FC<Props> = ({ data }) => {
+  const [t] = useTranslation("initialization");
+  const [startupKey, setStartupKey] = useState("");
+  const [userName, setUserName] = useState("scmadmin");
+  const [displayName, setDisplayName] = useState("SCM Administrator");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [passwordConfirmation, setPasswordConfirmation] = useState("");
+  const [userNameValidationError, setUserNameValidationError] = useState(false);
+  const [displayNameValidationError, setDisplayNameValidationError] = useState(false);
+  const [mailValidationError, setMailValidationError] = useState(false);
+  const [passwordValidationError, setPasswordValidationError] = useState(false);
+  const [passwordConfirmationValidationError, setPasswordConfirmationValidationError] = useState(false);
```

Comment:
I know you started this already some time ago... Wanna use `react-hook-form` here? It could simplify your code here a lot.